### ISSUE TITLE
Using the default value of timeout=0 when creating SSH connections might lead to threads stuck forever

### ISSUE THREAD

**Issue by romanek-adam** [sonyxperiadev/gerrit-events#97]
Hi, this is a follow-up of my investigation related to an incident me and my team was hit yesterday after our Gerrit instance outage.

I described the whole story in Jenkins' bug tracker as it was Jenkins' Gerrit Trigger plugin which was affected by the issue. See [JENKINS-33959](https://issues.jenkins-ci.org/browse/JENKINS-33959) and [my comment](https://issues.jenkins-ci.org/browse/JENKINS-33959?focusedCommentId=396177&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-396177) specifically.

I believe the root cause of the above issue comes from this library.

Long story short, when the timeout parameter is not specified the library uses the default value of 0 when creating SSH connections. This blocked the calling thread, potentially forever (it's still blocked 1 day after we noticed it).

As I mentioned in the Jenkins ticket:

> As a general thought setting timeouts to 0 is almost always a bad practice, unless you really want to wait forever. In practical applications it's usually better to let it timeout and simply retry.

**Comment by ModeSevenIndustrialSolutions** [comment]
Okay, I'm a bit surprised to see no additional comments on this report? We may be seeing problems with queues this in our production environment due to this!

**Comment by rsandell** [comment]
Yes, the default could be changed for a quick fix. But the trigger should probably be updated with a configurable connection timeout.
